dataland-frontend/src/utils/DataTableDisplay.ts

```diff
@@ -43,3 +43,83 @@ export type ReportingPeriodOfDataSetWithId = {
   dataId: string;
   reportingPeriod: string;
 };
+
+/**
+ * Adds click event listeners on DataTable row headers to expand and collapse row
+ * @param expandedRowsOnClick function passes the latest list of expanded row id's
+ * @param newExpandedRowsCallback function that returns the updated (after click) list of expanded row id's
+ * @returns the map of rows and their click handlers needed for unmounting
+ */
+export function mountRowHeaderClickEventListeners(
+  expandedRowsOnClick: () => string[],
+  newExpandedRowsCallback: (newExpandedRows: string[]) => void,
+): Map<Element, EventListener> {
+  const handlerMap: Map<Element | HTMLButtonElement, EventListener> = new Map();
+  let expandedRowGroups: string[] = [];
+
+  const t = setTimeout(() => {
+    const buttonAttributeName = "data-parent-id";
+    const rows = Array.from(document.querySelectorAll("[data-row-header-click]")).filter((el) => el);
+    const rowButtons = rows
+      .map((el: Element) => {
+        const button = el.parentNode?.querySelector('button[data-pc-section="rowgrouptoggler"]');
+        if (button) {
+          button?.setAttribute(buttonAttributeName, el.id);
+          return button;
+        }
+        return void 0;
+      })
+      .filter((button): button is HTMLButtonElement => !!button);
+
+    [...rows, ...rowButtons].forEach((el: Element | HTMLButtonElement) => {
+      let clickHandler: EventListener | null = (evt): void => {
+        if (!el.id) {
+          evt.stopImmediatePropagation();
+        }
```

Comment:
Just for my understanding:
If an element has no id, this means that the user probably clicked on the chevron.
Clicking on the chevron expands the row (via the primevue code) and therefore we don't want our event listener to do anything?

Reply:
Actually, this was a weird situation. Long story short, clicking on chevron emits both the chevron click and the row header beneath it. It still worked fine, but it ticked me off that 2 events were being emitted.
Now clicking on chevron it tops the chevron's event, but bubbles down to the parent row which fires its own event.

Reply:
So our custom event, which is triggered by a click anywhere on the row, is surpressed, if the chevron (button) is clicked. Correct?

Reply:
I believe it is the other way around... but in general yes, one of them is suppressed and only one element emits the event.

Reply:
Somehow this does not work properly.   I added some console logs and clicked the chevron many times on different rows, and this happened (see console):

![image](https://github.com/d-fine/Dataland/assets/99255296/7cea006b-82d7-433a-8a82-965ab7fe7a2e)


Some empty string "" are being added to the expandedRowGroups list.

Reply:
I did similar tests before and got only one emit. I'll check. Perhaps something creeped during latest changes.

Reply:
I added an additional safeguard to stop the empty string (we removed the value from the attribute earlier).
I re-tested this with a number of console.logs and seems to be working fine.

Reply:
Ok will check this out now.  Thx